src/v8/inspector/server.rs

```diff
@@ -0,0 +1,882 @@
+                    log::trace!("Parsed script: {parsed_script:?}");
+                }
+            }
+            log::trace!("Responding with string {s}.");
+            match websocket.lock() {
+                Ok(mut websocket) => match websocket.0.send(Message::Text(s)) {
+                    Ok(_) => log::trace!("Responded with string successfully."),
+                    Err(e) => log::error!("Couldn't send a websocket message to the client: {e}"),
+                },
+                Err(e) => log::error!("Couldn't lock the socket: {e}"),
+            }
+        };
+
+        let on_wait_frontend_message_on_pause = move |raw: *mut crate::v8_c_raw::bindings::v8_inspector_c_wrapper| -> std::os::raw::c_int {
+            // When this is returned, the callback will be called again,
+            // if no error occured.
+            const CONTINUE_WAITING: std::os::raw::c_int = 0;
+            // Returning this would result in stopping the wait.
+            const STOP_WAITING: std::os::raw::c_int = 1;
+
+            let string;
+
+            loop {
+                match web_socket.lock() {
+                    Ok(mut websocket) => match websocket.read_next_message() {
+                        Ok(message) => {
+                            log::trace!("[OnWait] Read the message: {message:?}");
+
+                            string = match std::ffi::CString::new(message) {
+                                Ok(string) => string,
+                                _ => continue,
+                            };
+                            break;
+                        }
+                        Err(e) => if e.kind() == std::io::ErrorKind::ConnectionAborted {
+                            return CONTINUE_WAITING;
+                        }
+                    },
+                    Err(e) => {
+                        log::error!("The WebSocketServer mutex is poisoned: {e:?}");
+                        return CONTINUE_WAITING;
+                    },
+                }
+            }
+
+            unsafe {
+                crate::v8_c_raw::bindings::v8_InspectorDispatchProtocolMessage(
+                    raw,
+                    string.as_ptr(),
+                )
+            }
+
+            STOP_WAITING
+        };
+
+        InspectorCallbacks {
+            on_response,
+            on_wait_frontend_message_on_pause,
+        }
+    }
+
+    /// Creates a new debugger to be used with the provided [Inspector].
+    /// Starts a web socket debugging session using [WebSocketServer].
+    ///
+    /// Once the connection is accepted and the inspector starts, the
+    /// function returns.
+    ///
+    /// After the function returns, to start the debugging main loop,
+    /// one needs to call the [`Self::process_messages`].
+    /// method.
+    ///
+    /// # Notes
+    ///
+    /// This method requires a [`V8IsolateScope`].
+    pub fn new(
+        web_socket: WebSocketServer,
+        inspector: Arc<Inspector>,
```

Comment:
Why it can not owned the inspector? Can you start multiple debugging sessions on the same inspector?

Reply:
Keep it like this till we review the RedisGears code